homeassistant/components/binary_sensor/hive.py

```diff
@@ -0,0 +1,66 @@
+"""
+Support for the Hive devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hive/
+"""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.hive import DATA_HIVE
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEVICETYPE_DEVICE_CLASS = {'motionsensor': 'motion',
+                           'contactsensor': 'opening'}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Hive sensor devices."""
+    session = hass.data.get(DATA_HIVE)
+
+    add_devices([HiveBinarySensorEntity(session, discovery_info)])
+
+
+class HiveBinarySensorEntity(BinarySensorDevice):
+    """representation of a Hive binary sensor."""
+
+    def __init__(self, hivesession, hivedevice):
+        """Initialize the hive sensor."""
+        self.node_id = hivedevice["Hive_NodeID"]
+        self.node_name = hivedevice["Hive_NodeName"]
+        self.device_type = hivedevice["HA_DeviceType"]
+        self.node_device_type = hivedevice["Hive_DeviceType"]
+        self.session = hivesession
+
+    def handle_update(self, updatesource):
+        """Handle the new update request."""
+        if '{}.{}'.format(self.device_type, self.node_id) not in updatesource:
+            self.schedule_update_ha_state()
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return DEVICETYPE_DEVICE_CLASS.get(self.node_device_type)
+
+    @property
+    def should_poll(self):
+        """No polling needed for the Hive binary sensor."""
+        return False

```

Comment:
Shouldn't this entity class poll like the other entities?

Reply:
I don't know, please advise.
its data is not really live, it is retrieved from the Hive API along with the rest of the device data, by default every 2 minutes. It saying get more data will just get from the cached instance data, if the cached instance data is over 2 minutes old then it will get new data from the Hive API. 

So maybe whilst explaining I have come to the conclusion that it needs to poll = true to it gets new data as opposed to something give it data?

Reply:
:+1:

Reply:
thanks. updated